core/templates/dev/head/domain/topic/TopicUpdateService.js

```diff
@@ -245,6 +298,12 @@ oppia.factory('TopicUpdateService', [
             if (backendChangeDict.subtopic_id > subtopicId) {
               backendChangeDict.subtopic_id--;
             }
+            if (backendChangeDict.old_subtopic_id > subtopicId) {
```

Comment:
Again ... no filtering on type of change?

Reply:
Whatever the command is, any reference to a subtopic id, following the condition given, should be decremented here. So, if future change commands with different fields that references subtopic id is created also, it should be decremented here.

Reply:
Then, shall I check whether backendChangeDict has the attributes instead? Since, any future commands which would have these fields would also require it.

Reply:
Yeah, something like backendChangeDict.hasOwnProperty(old_subtopic_id).

As an aside, this is the point where I wish we were dealing with change domain objects in the frontend rather than the backend change dict directly. It would be so much cleaner for the object itself to have a method that handles this.

Reply:
Done.